src/pyqasm/maps/gates.py

```diff
@@ -1,1335 +1,1221 @@
-    # not sure if we wanna hande array bit assignments too.
-    # For now, we only cater to single bit assignment.
-    if openqasm_type == BitType:
-        return rhs_value != 0
-    if openqasm_type == AngleType:
-        return rhs_value  # not sure
-
-
-# IEEE 754 Standard for floats
-# https://openqasm.com/language/types.html#floating-point-numbers
-LIMITS_MAP = {"float_32": 1.70141183 * (10**38), "float_64": 10**308}
-
-CONSTANTS_MAP = {
-    "π": 3.141592653589793,
-    "pi": 3.141592653589793,
-    "ℇ": 2.718281828459045,
-    "euler": 2.718281828459045,
-    "τ": 6.283185307179586,
-    "tau": 6.283185307179586,
-}
-
-VARIABLE_TYPE_MAP = {
-    BitType: bool,
-    IntType: int,
-    UintType: int,
-    BoolType: bool,
-    FloatType: float,
-    ComplexType: complex,
-    # AngleType: None,  # not sure
-}
-
-# Reference: https://openqasm.com/language/types.html#allowed-casts
-VARIABLE_TYPE_CAST_MAP = {
-    BoolType: (int, float, bool, np.int64, np.float64, np.bool_),
-    IntType: (bool, int, float, np.int64, np.float64, np.bool_),
-    BitType: (bool, int, np.int64, np.bool_),
-    UintType: (bool, int, float, np.int64, np.uint64, np.float64, np.bool_),
-    FloatType: (bool, int, float, np.int64, np.float64, np.bool_),
-    AngleType: (float, np.float64),
-}
-
-ARRAY_TYPE_MAP = {
-    BitType: np.bool_,
-    IntType: np.int64,
-    UintType: np.uint64,
-    FloatType: np.float64,
-    ComplexType: np.complex128,
-    BoolType: np.bool_,
-    AngleType: np.float64,
-}
-
-# Reference : https://openqasm.com/language/types.html#arrays
-MAX_ARRAY_DIMENSIONS = 7
-
-# Reference : https://openqasm.com/language/classical.html#the-switch-statement
-# Paragraph 14
-SWITCH_BLACKLIST_STMTS = {
-    QubitDeclaration,
-    ClassicalDeclaration,
-    SubroutineDefinition,
-    QuantumGateDefinition,
-}
-
-SUPPORTED_QASM_VERSIONS = {"3.0", "3", "2", "2.0"}
-
-QUANTUM_STATEMENTS = (QuantumGate, QuantumBarrier, QuantumReset, QuantumMeasurementStatement)
+# Copyright (C) 2024 qBraid
+#
+# This file is part of pyqasm
+#
+# Pyqasm is free software released under the GNU General Public License v3
+# or later. You can redistribute and/or modify it under the terms of the GPL v3.
+# See the LICENSE file in the project root or <https://www.gnu.org/licenses/gpl-3.0.html>.
+#
+# THERE IS NO WARRANTY for pyqasm, as per Section 15 of the GPL v3.
+
+# pylint: disable=too-many-lines
+
+"""
+Module mapping supported QASM gates to lower level gate operations.
+
+"""
+
+
+from typing import Callable, Union
+
+import numpy as np
+from openqasm3.ast import (
+    FloatLiteral,
+    Identifier,
+    IndexedIdentifier,
+    QuantumGate,
+    QuantumPhase,
+)
+
+from pyqasm.elements import InversionOp
+from pyqasm.exceptions import ValidationError
+from pyqasm.linalg import kak_decomposition_angles
+from pyqasm.maps import OperatorFunction
+from pyqasm.maps.expressions import CONSTANTS_MAP
+
+
+OPERATOR_MAP: dict[str, OperatorFunction] = {

```

Comment:
This should be moved to `expressions.py`

Reply:
Done